rapids-cmake/test/add.cmake

```diff
@@ -0,0 +1,149 @@
+#=============================================================================
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+rapids_test_add
+---------------
+
+.. versionadded:: v23.04.00
+
+States how many GPUs and what percent of each a test requires.
+
+  .. code-block:: cmake
+
+    rapids_test_add(NAME <name> COMMAND <target|command> [<args>...]
+                    [GPUS <N> [PERCENT <value>]]
+                    [INSTALL_COMPONENT_SET <set>]
+                    [WORKING_DIRECTORY <dir>])
+
+Add a test called `<name>` which will be executed with a given GPU
+resource allocation.
+
+When combined with :cmake:command:`rapids_test_init` informs CTest what
+resources should be allocated to a test so that when testing in parallel
+oversubscription doesn't occur. Without this information user execution of
+CTest with high parallel levels will cause multiple tests to run on the
+same GPU and quickly exhaust all memory.
+
+``COMMAND``
+  Specify the test command-line. If <command> specifies an executable target
+  (created by add_executable()) it will automatically be replaced by the location of the
+  executable created at build time.
```

Comment:
It would be helpful to indicate that COMMAND isn't just a single executable but can be the entire command. I don't know the best way to phrase this, but I mention it because I was initially confused by the way that args are separated out from the command below.

Reply:
Made this clearer.